### DISCUSSION TITLE
cookie not visible in the response when inspected via browser console. Intended?

### DISCUSSION THREAD

**Discussion by s-bose** [fastapi/fastapi#6575] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.post("/login")
def login(request: Request, cred: UserLogin, db: Session = Depends(get_db)):

    user_email = cred.email
    user_pass = cred.master_pwd

    password = get_hash(user_pass)
    if (res := crud_user.get_user_by_email(db, cred.email)) is None:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND, detail="email not found"
        )

    if (user := crud_user.get_user(db, email=user_email, password=password)) is None:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND, detail="invalid email/password"
        )

    # token expires after X minutes
    time_expires = timedelta(minutes=ACCESS_TOKEN_EXPIRE_MINUTES)
    payload = { "user_info": user.dict(exclude_unset=True) }

    access_token = create_access_token(data=payload, expires_delta=time_expires)

    response = JSONResponse(content=payload)
    response.set_cookie(
        key="access_token",
        value=access_token,
        max_age=time_expires.total_seconds(),
    )
    return payload
```

### Description

When returning just the payload, if I open the browser console I see that the response header does not contain any information about any cookie, but the website has the cookie set if I view site information.
However, when I return the `response` instead of just the payload, the response header contains an entry `set-cookie` that contains `access-token`.
Is this intended? 
I am asking this because in the docs it says the following:

> FastAPI will use that temporal response to extract the cookies (also headers and status code), and will put them in the final response that contains the value you returned, filtered by any `response_model`.

And since I didn't specify any `response_model` it should have populated the response headers with the cookie info the same way as returning the response object directly. but that is not happening.

### Operating System

Linux

### Operating System Details

Pop OS. 21.04.

### FastAPI Version

0.63.0

### Python Version

3.9.6

**Comment by frankie567** [comment, marked as answer]
The problem here is that you define a new `Response` object but don't do anything with it. So it doesn't do anything in the example you show here.

You should either return this response object or, better, inject the `Response` object in argument and directly set the cookie. FastAPI will take care of it:

```py
@router.post("/login")
def login(request: Request, response: Response, cred: UserLogin, db: Session = Depends(get_db)):
    # ... your code ...

    response.set_cookie(
        key="access_token",
        value=access_token,
        max_age=time_expires.total_seconds(),
    )

    return payload
```

With this method, FastAPI will also honor the `status_code` and `response_model` arguments you might set in the decorator.

Ref: https://fastapi.tiangolo.com/advanced/response-cookies/#response-cookies

**Comment by s-bose** [comment]
Thank you. I missed that bit of detail from the docs. This works.
